src/Microsoft.NetCore.Analyzers/UnitTests/ImmutableCollections/DoNotCallToImmutableCollectionOnAnImmutableCollectionValueTests.Fixer.cs

```diff
@@ -29,5 +32,199 @@ protected override CodeFixProvider GetCSharpCodeFixProvider()
         {
             return new CSharpDoNotCallToImmutableCollectionOnAnImmutableCollectionValueFixer();
         }
+
+        public static readonly TheoryData<string> CollectionNames_Arity1 = new TheoryData<string>
+        {
+            nameof(ImmutableArray),
+            nameof(ImmutableHashSet),
+            nameof(ImmutableList),
+            nameof(ImmutableSortedSet)
+        };
+
+        public static readonly TheoryData<string> CollectionNames_Arity2 = new TheoryData<string>
+        {
+            nameof(ImmutableDictionary),
+            nameof(ImmutableSortedDictionary)
+        };
+
+        [Theory]
+        [MemberData(nameof(CollectionNames_Arity1))]
+        public void CA2009_Arity1_CSharp(string collectionName)
+        {
+            var initial = $@"
+using System.Collections.Generic;
+using System.Collections.Immutable;
+
+static class Extensions
+{{
+     public static {collectionName}<TSource> To{collectionName}<TSource>(this IEnumerable<TSource> items)
+     {{
+         return default({collectionName}<TSource>);
+     }}
+}}
+
+class C
+{{
+    public void M(IEnumerable<int> p1, List<int> p2, {collectionName}<int> p3)
+    {{
+        var x = p1.To{collectionName}().To{collectionName}();
```

Comment:
Please also add some tests of calling ToImmutableXYZ in static method form. e.g. something like `ImmutableExtensions.ToToImmutableArray(p1)`

Reply:
@genlu I was only able to accomplish this in C# test cases. Looks like the analyzer doesn't report this kind of invocations in VB.

How should we address that? It's an analyzer bug, so it sounds like it's beyond the scope of this PR.

Reply:
I agree. Could you please file an issue for VB analyzer?

Reply:
@genlu here you go - https://github.com/dotnet/roslyn-analyzers/issues/2283

Thank you very much for helping me getting this in. Can I contact you for future PR-s too?

Reply:
No problem. Feel free to tag me for code reviews and bugs.